python/shark_turbine/aot/support/ir_utils.py

```diff
@@ -221,3 +228,21 @@ def emit_return(self, *ir_values: Value):
             ftype = FunctionType.get(ftype.inputs, value_types)
             self.func_op.attributes["function_type"] = TypeAttr.get(ftype)
             assert self.func_op.verify(), "Created function is invalid"
+
+
+###############################################################################
+# Helpers
+###############################################################################
+
+
+def build_index_attribute(value: int) -> IntegerAttr:
+    return IntegerAttr.get(IndexType.get(), value)
+
+
+def build_index_value(value: int) -> Value:
+    attr = build_index_attribute(value)
+    return arith_d.ConstantOp(attr.type, attr).result
```

Comment:
I don't know why the type would need to be specified, arith.constant should have type inference.

Reply:
Someone overrode it to be that way in an extension. Indeed, the ODS generated one is "better" than the hand-coded one now. But compatibility, I guess :/  (took me a minute to figure out why it was not working)